File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities/Shared/Layouts.vb
Option Explicit On
Option Strict On
Option Infer Off

Imports Autodesk.AutoCAD.ApplicationServices
Imports Autodesk.AutoCAD.DatabaseServices

Public Class Layouts
    Public Shared Function CreateTab(doc As Document, originalTabName As String) As ObjectId
        Dim actualTabName As String = originalTabName
        Dim db As Database = doc.Database
        Using lock As DocumentLock = doc.LockDocument()
            Using t As Transaction = db.TransactionManager.StartTransaction()

                Dim existingId As ObjectId = LayoutManager.Current.GetLayoutId(actualTabName)
                Dim count As Integer = 1
                While (Not existingId.IsNull)
                    actualTabName = originalTabName + "-" + count.ToString
                    existingId = LayoutManager.Current.GetLayoutId(actualTabName)
                    count += 1
                End While
                Dim id As ObjectId = LayoutManager.Current.CreateLayout(actualTabName)
                Dim dbObj As DBObject = t.GetObject(id, OpenMode.ForWrite)
                Dim layout As Layout = CType(dbObj, Layout)
                LayoutManager.Current.CurrentLayout = layout.LayoutName
                't.AddNewlyCreatedDBObject(dbObj, True)
                t.Commit()
                Return id
            End Using
        End Using

        Return Nothing
    End Function

    Public Shared Sub SetCurrentTab(doc As Document, tab As String)
        Dim db As Database = doc.Database
        Dim number2 As Integer = db.TransactionManager.NumberOfActiveTransactions()
        Using lock As DocumentLock = doc.LockDocument()
            Using tm As Transaction = db.TransactionManager.StartTransaction()
                Dim existingId As ObjectId = LayoutManager.Current.GetLayoutId(tab)
                If Not existingId.IsNull Then
                    LayoutManager.Current.CurrentLayout = tab
                    tm.Commit()
                End If

            End Using
        End Using
    End Sub
End Class

File: cadwiki-nuget/cadwiki.AC/Utilities/Layouts.cs

using Autodesk.AutoCAD.ApplicationServices;
using Autodesk.AutoCAD.DatabaseServices;

namespace cadwiki.AC.Utilities
{

    public class Layouts
    {
        public static ObjectId CreateTab(Document doc, string originalTabName)
        {
            string actualTabName = originalTabName;
            var db = doc.Database;
            using (var @lock = doc.LockDocument())
            {
                using (var t = db.TransactionManager.StartTransaction())
                {

                    var existingId = LayoutManager.Current.GetLayoutId(actualTabName);
                    int count = 1;
                    while (!existingId.IsNull)
                    {
                        actualTabName = originalTabName + "-" + count.ToString();
                        existingId = LayoutManager.Current.GetLayoutId(actualTabName);
                        count += 1;
                    }
                    var id = LayoutManager.Current.CreateLayout(actualTabName);
                    var dbObj = t.GetObject(id, global::Autodesk.AutoCAD.DatabaseServices.OpenMode.ForWrite);
                    Layout layout = (Layout)dbObj;
                    LayoutManager.Current.CurrentLayout = layout.LayoutName;
                    // t.AddNewlyCreatedDBObject(dbObj, True)
                    t.Commit();
                    return id;
                }
            }

            return default;
        }

        public static void SetCurrentTab(Document doc, string tab)
        {
            var db = doc.Database;
            int number2 = db.TransactionManager.NumberOfActiveTransactions;
            using (var @lock = doc.LockDocument())
            {
                using (var tm = db.TransactionManager.StartTransaction())
                {
                    var existingId = LayoutManager.Current.GetLayoutId(tab);
                    if (!existingId.IsNull)
                    {
                        LayoutManager.Current.CurrentLayout = tab;
                        tm.Commit();
                    }

                }
            }
        }
    }
}